pyproject.toml

```diff
@@ -67,10 +67,11 @@ scripts_are_modules = true
 warn_unused_configs = true
 show_error_codes = true
 allow_redefinition = true
+check_untyped_defs = true
 
 [tool.pylint.messages_control]
 max-line-length = 90
-disable = []
+disable = ["duplicate-code"]
```

Comment:
Why should duplicating code be allowed?

Reply:
```shell
************* Module ci_cd.utils
ci_cd/utils.py:1:0: R0801: Similar lines in 2 files
==ci_cd.tasks.api_reference_docs:[18:114]
==ci_cd.tasks.update_deps:[16:42]
if TYPE_CHECKING:  # pragma: no cover
    from invoke import Context, Result


LOGGER = logging.getLogger(__file__)
LOGGER.setLevel(logging.DEBUG)


@task(
    help={
        "package-dir": (
            "Relative path to a package dir from the repository root, "
            "e.g., 'src/my_package'. This input option can be supplied multiple times."
        ),
        "pre-clean": "Remove the 'api_reference' sub directory prior to (re)creation.",
        "pre-commit": (
            "Whether or not this task is run as a pre-commit hook. Will return a "
            "non-zero error code if changes were made."
        ),
        "root-repo-path": (
            "A resolvable path to the root directory of the repository folder."
        ),
        "docs-folder": (
            "The folder name for the documentation root folder. "
            "This defaults to 'docs'."
        ),
        "unwanted-folder": (
            "A folder to avoid including into the Python API reference documentation. "
            "Note, only folder names, not paths, may be included. Note, all folders "
            "and their contents with this name will be excluded. Defaults to "
            "'__pycache__'. This input option can be supplied multiple times."
        ),
        "unwanted-file": (
            "A file to avoid including into the Python API reference documentation. "
            "Note, only full file names, not paths, may be included, i.e., filename + "
            "file extension. Note, all files with this names will be excluded. "
            "Defaults to '__init__.py'. This input option can be supplied multiple "
            "times."
        ),
        "full-docs-folder": (
            "A folder in which to include everything - even those without "
            "documentation strings. This may be useful for a module full of data "
            "models or to ensure all class attributes are listed. This input option "
            "can be supplied multiple times."
        ),
        "full-docs-file": (
            "A full relative path to a file in which to include everything - even "
            "those without documentation strings. This may be useful for a file full "
            "of data models or to ensure all class attributes are listed. This input "
            "option can be supplied multiple times."
        ),
        "special-option": (
            "A combination of a relative path to a file and a fully formed "
            "mkdocstrings option that should be added to the generated MarkDown file. "
            "The combination should be comma-separated. Example: "
            "'my_module/py_file.py,show_bases:false'. Encapsulate the value in double "
            'quotation marks (") if including spaces ( ). Important: If multiple '
            "package-dir options are supplied, the relative path MUST include/start "
            "with the package-dir value, e.g., "
            "'\"my_package/my_module/py_file.py,show_bases: false\"'. This input "
            "option can be supplied multiple times. The options will be accumulated "
            "for the same file, if given several times."
        ),
        "relative": (
            "Whether or not to use relative Python import links in the API reference "
            "markdown files."
        ),
        "debug": "Whether or not to print debug statements.",
    },
    iterable=[
        "package_dir",
        "unwanted_folder",
        "unwanted_file",
        "full_docs_folder",
        "full_docs_file",
        "special_option",
    ],
)
def create_api_reference_docs(  # pylint: disable=too-many-locals,too-many-branches,too-many-statements,line-too-long
    context,
    package_dir,
    pre_clean=False,
    pre_commit=False,
    root_repo_path=".",
    docs_folder="docs",
    unwanted_folder=None,
    unwanted_file=None,
    full_docs_folder=None,
    full_docs_file=None,
    special_option=None,
    relative=False,
    debug=False,
):
    """Create the Python API Reference in the documentation."""
    if TYPE_CHECKING:  # pragma: no cover
        context: "Context" = context  # type: ignore[no-redef] (duplicate-code)
```

The above is from `api_reference_docs`. The following are the lines 16:42 for `update_docs`:

```shell
if TYPE_CHECKING:  # pragma: no cover
    from invoke import Context, Result


LOGGER = logging.getLogger(__file__)
LOGGER.setLevel(logging.DEBUG)


@task(
    help={
        "fail-fast": (
            "Fail immediately if an error occurs. Otherwise, print and ignore all "
            "non-critical errors."
        ),
        "root-repo-path": (
            "A resolvable path to the root directory of the repository folder."
        ),
        "pre-commit": "Whether or not this task is run as a pre-commit hook.",
    }
)
def update_deps(  # pylint: disable=too-many-branches,too-many-locals,too-many-statements
    context, root_repo_path=".", fail_fast=False, pre_commit=False
):
    """Update dependencies in specified Python package's `pyproject.toml`."""
    if TYPE_CHECKING:  # pragma: no cover
        context: "Context" = context  # type: ignore[no-redef]
```

So, essentially, it doesn't consider decorators for some reason. Or rather, it doesn't consider parameters/arguments in decorators.

I'd argue this code duplication cannot be de-duplicated in a reasonable manner and should therefore be silenced/disabled.

Reply:
I can possibly try to add the disable pylint statement at the top of those two files instead, as a compromise? :)

Reply:
yes, that is maybe better if possible.